collection/list/list.go

```diff
@@ -0,0 +1,43 @@
+package list
```

Comment:
I'd really like to have this in `qri/api/util` as there's already some of the required logic present and would be good to keep it together and source it from a central place for other endpoints besides just collection.

Reply:
I don't think this belongs in `api/util`. Trying out a new top-level `params` package, would love your thoughts

Reply:
Huh, honestly, no clue what should be best. This feels a lot like going full circle on some things.

At the start of the API refactor I went a similar route where I started condensing some of these "generic" params and some others into `lib/params.go` which we then decided not to do. We wanted to have each params struct have it's own in it's own file (and potentially even split wherever possible even at the expense of duplicating code; think Dustin was dealing with that part specifically).

I'd say we just pick one setup and stick with it, either has it's own mini tradeoffs none of which are worth fussing around too much. My vote would be in `api/util` or a standalone package, I'd hate to have very large imports (and potential circular deps) just because it lives in a large package. Other than that I'm fine with anything.